### DISCUSSION TITLE
Build Fails on Windows 11 with slint Using Skia Renderer Due to Metal Feature

### DISCUSSION THREAD

**Discussion by gtravs** [slint-ui/slint#7067] [Q&A]
### Bug Description

When using slint with the Skia renderer on Windows 11, the build fails due to unresolved imports and missing symbols related to the Metal API. It seems that the metal feature for skia-safe is being enabled incorrectly during the build process, even though Metal is not supported on Windows.

 --> .cargo\registry\src\index.crates.io-6f17d22bba15001f\skia-safe-0.78.2\src\gpu\ganesh\mtl\backend_context.rs:8:33
  |
8 | use skia_bindings::{self as sb, GrMtlBackendContext};
  |                                 ^^^^^^^^^^^^^^^^^^^
  |                                 |
  |                                 no `GrMtlBackendContext` in the root
  |                                 help: a similar name exists in the module: `GrD3DBackendContext`

error[E0432]: unresolved imports `skia_bindings::GrMtlSurfaceInfo`, `skia_bindings::GrMtlTextureInfo`
 --> \.cargo\registry\src\index.crates.io-6f17d22bba15001f\skia-safe-0.78.2\src\gpu\ganesh\mtl\types.rs:3:33
  |
3 | use skia_bindings::{self as sb, GrMtlSurfaceInfo, GrMtlTextureInfo};
  |                                 ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^ no `GrMtlTextureInfo` in the root
  |                                 |
  |                                 no `GrMtlSurfaceInfo` in the root
  |
help: a similar name exists in the module
  |
3 | use skia_bindings::{self as sb, GrGLSurfaceInfo, GrMtlTextureInfo};
  |                                 ~~~~~~~~~~~~~~~
help: a similar name exists in the module
  |
3 | use skia_bindings::{self as sb, GrMtlSurfaceInfo, GrMockTextureInfo};
  |                                                   ~~~~~~~~~~~~~~~~~

error[E0432]: unresolved import `skia_bindings::GrMTLHandle`
  --> \.cargo\registry\src\index.crates.io-6f17d22bba15001f\skia-safe-0.78.2\src\gpu\ganesh\mtl\types.rs:10:9
   |
10 | pub use skia_bindings::GrMTLHandle as Handle;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `GrMTLHandle` in the root

error[E0432]: unresolved import `skia_bindings::GrMTLPixelFormat`
  -->\.cargo\registry\src\index.crates.io-6f17d22bba15001f\skia-safe-0.78.2\src\gpu\ganesh\mtl\types.rs:11:9
   |
11 | pub use skia_bindings::GrMTLPixelFormat as PixelFormat;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `GrMTLPixelFormat` in the root

error[E0432]: unresolved import `skia_bindings::GrMTLStorageMode`
  --> \.cargo\registry\src\index.crates.io-6f17d22bba15001f\skia-safe-0.78.2\src\gpu\ganesh\mtl\types.rs:12:9
   |
12 | pub use skia_bindings::GrMTLStorageMode as StorageMode;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `GrMTLStorageMode` in the root

error[E0432]: unresolved import `skia_bindings::GrMTLTextureUsage`
  --> \.cargo\registry\src\index.crates.io-6f17d22bba15001f\skia-safe-0.78.2\src\gpu\ganesh\mtl\types.rs:13:9
   |
13 | pub use skia_bindings::GrMTLTextureUsage as TextureUsage;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `GrMTLTextureUsage` in the root

error[E0425]: cannot find function, tuple struct or tuple variant `C_GrMtlBackendContext_Destruct` in crate `sb`
  --> \.cargo\registry\src\index.crates.io-6f17d22bba15001f\skia-safe-0.78.2\src\gpu\ganesh\mtl\backend_context.rs:15:22
   |
15 |         unsafe { sb::C_GrMtlBackendContext_Destruct(self) }
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `C_GrBackendFormat_destruct`
   |

### Reproducible Code (if applicable)

```slint
[dependencies]
slint = { version = "1.8.0", default-features = false, features = [
    "compat-1-2",
    "std",
    "backend-winit",
    "renderer-skia-opengl"
] }
winit = "0.30"
i-slint-backend-winit = "1.8.0"
[build-dependencies]
slint-build = "1.8.0"
```

### Environment Details

Environment
Operating System: Windows 11
Rust Version: 1.78.0 (MSVC toolchain)
slint Version: 1.8
skia-safe Version: 0.78.2
Toolchain: x86_64-pc-windows-msvc

### Product Impact

I am building a cross-platform GUI application using Slint with the Skia renderer. The application targets Windows as one of the primary platforms, alongside other platforms like Linux and macOS. The project requires hardware-accelerated rendering for interactive and graphical content.

This issue is a blocker for my project. The build fails on Windows 11 due to the metal feature being incorrectly enabled during the build process, which is unsupported on Windows. This prevents me from proceeding with development or deployment for Windows users, which is a critical platform for my product. Resolving this issue is essential for enabling compatibility and ensuring a smooth development experience.

**Comment by tronical** [comment]
This looks like it's trying to compile the metal backend also on Windows - that's going to fail.

Could you try and see with `cargo tree -e features` on your app if the `metal` feature is enabled for `skia-safe` somehow?
